Models/CoreBitcoinManager.swift

```diff
@@ -10,22 +10,19 @@ import Foundation
 import BlockchainCertificates
 
 class CoreBitcoinManager : BitcoinManager {
-    func address(for certificate: Certificate, on chain: String) -> String? {
+    func address(for message: String, for signature: String, on chain: String) -> String? {
```

Comment:
maybe `with` signature? I like it like this because it's a more standard, reusable bitcoin message signing technique. It asks for the minimum params needed.